Wagtail 7.1 maintenance

nickmoreton opened the pull request:
This pull request updates the project to support newer versions of Python, Django, and Wagtail, and refactors several views and migrations for improved compatibility and maintainability. 

**Dependency and compatibility updates:**

* Updated supported versions in `.github/workflows/test.yml` and `pyproject.toml` to Python 3.13, Django 5.2, and Wagtail 7.x, and adjusted test matrix accordingly. 

**View refactoring for Wagtail version support:**

* Refactored `src/wagtail_bynder/views/document.py` and `src/wagtail_bynder/views/image.py` to remove legacy support for Wagtail <6.3

**Testing improvements:**

* Updated test assertions in `tests/test_image_chooser_views.py` 

**Migration and model restructuring:**

* Reworked `tests/testapp/migrations/0001_initial.py` 

One could consider #37 to be not required and #38 to be included.

nickmoreton committed:
Revise the test matrix and dependencies

nickmoreton committed:
Recreated the test migration

nickmoreton committed:
Remove cmpatibility code for Wagtail < 6.3

nickmoreton committed:
Update test assertions to be less brittle

zerolab commented on tests/test_image_chooser_views.py:
```diff
@@ -40,18 +40,15 @@ def test_creates_image_if_asset_id_not_recognised(self):
         create_object_mock.assert_called_once()
         self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertEqual(response_json["step"], "chosen")
```
Let's keept it at full dict. Fewer surprises.

nickmoreton committed:
Revert "Update test assertions to be less brittle"

This reverts commit 5df2748784e03e820c8bceb8919393b6d96f6dd2.

nickmoreton committed:
Add conditional inclusion of 'default_alt_text' in test assertions based on Wagtail version.

Technically this is a wagtail 6.4+ compatibility fix. But as 6.4 isn't a supported or tested version then 7.0+ is OK.

nickmoreton committed:
Update telepath import location
